### PR TITLE
Extend cluster definition to support spot instances

### PR DESCRIPTION
**Pull request by marians**
The goal of this PR is to add to `gsctl create cluster` the missing support for node pool attributes that got added witt spot instances, in the case that a YAML cluster definition is used.

### CODE REVIEW

File: commands/create/cluster/command_test.go [hunk 1]
```diff
@@ -215,19 +215,19 @@ func Test_CreateClusterSuccessfully(t *testing.T) {
 					ReleaseVersion:    "1.2.3",
 					AvailabilityZones: 1,
 					Workers: []types.NodeDefinition{
-						types.NodeDefinition{
+						{
 							Memory:  types.MemoryDefinition{SizeGB: 2},
```
Comment (marians) [inline commands/create/cluster/command_test.go:219]: These are some unrelated boyscouting changes to quiet golint

**Review by axbarsan — APPROVED** [review]

**Review by pipo02mix — APPROVED** [review]
LGTM
